controller/tasks_manager.go

```diff
@@ -404,6 +404,15 @@ func (tm *TasksManager) checkApply(ctx context.Context, d driver.Driver, retry,
 	return rendered, nil
 }
 
+// EnableTestMode is a helper for testing which tasks were triggered and
+// executed. Callers of this method must consume from TaskNotify channel to
+// prevent the buffered channel from filling and causing a dead lock.
+func (tm *TasksManager) EnableTestMode() <-chan string {
```

Comment:
So this method sets the channels for `tm.taskNotify`. I know that this method is used for testing, but there isn't anything being done here that feels like it couldn't be used for production. What I mean is, if in the future, if we wanted to have some entity that monitored live task runs, this method could be used for that.

Maybe a name that doesn't include Test would be good? `SetNotificationChannels` or `EnableTaskNotify`?

Reply:
Good idea, I could rename this to something like `EnableTaskRunNotify` (in a subsequent PR I added something like a `EnableTaskDeleteNotify`). If it's ok with you I'd like to do this renaming in a separate PR as it will create conflicts with another PR I already opened.

Reply:
that works for me!